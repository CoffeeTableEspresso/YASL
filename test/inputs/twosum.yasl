##10 + 4 == 14\nnot found\n
/* problem: given an unsorted list of unique integers `nums` and a target integer `target`,
 * write a function that returns the two numbers in `nums` that sum to target. There may be
 * multiple pairs that sum to `target`, in which case you may return any such pair.
 */

const fn twosum(const nums, const target) {
    x := { target - num: num for num <- nums }
    for y <- nums {
        if x[y] {
            return "#{x[y]} + #{y} == #{target}"
        }
    }
    return 'not found'
}


echo twosum([1, 2, 5, 4, 10], 14)
echo twosum([1, 2, 5, 4, 10], 16)

