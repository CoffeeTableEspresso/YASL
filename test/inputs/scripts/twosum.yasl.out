10 + 4 == 14
not found
